kkm/kkm_main.c

```diff
@@ -768,17 +768,21 @@ static int kkm_get_native_cpuid(unsigned long arg)
 	if ((ret_val = kkm_to_user(&(((struct kkm_cpuid *)arg)->entries[0]), ec,
 				   ec_entries_size)) != 0) {
 		ret_val = -EFAULT;
+		goto error;
 	}
 
 	kcpuid.reserved = 0;
 	if ((ret_val = kkm_to_user((void *)arg, &kcpuid,
 				   sizeof(struct kkm_cpuid))) != 0) {
 		ret_val = -EFAULT;
```

Comment:
ret_val already has the value -EFAULT since kkm_to_user() returns that if an error is found.
Assigning -EFAULT to ret_val again is redundant.
But it is harmless if you like this sort of belt and suspenders approach.
Could also assign that value several times to let the compiler know you are serious.